naga/tests/snapshots.rs

```diff
@@ -331,18 +342,27 @@ fn check_targets(
                 debug_info,
                 &params.spv,
                 params.bounds_check_policies,
+                &params.pipeline_constants,
             );
         }
     }
     #[cfg(all(feature = "deserialize", feature = "msl-out"))]
     {
         if targets.contains(Targets::METAL) {
+            let mut pipeline_options = params.msl_pipeline.clone();
+            let old = std::mem::replace(
+                &mut pipeline_options.constants,
+                params.pipeline_constants.clone(),
+            );
+            if !old.is_empty() {
+                panic!("Supply pipeline constants via pipeline_constants instead of msl_pipeline.constants!");
+            }
```

Comment:
This could be simpler:

```suggestion
            if !params.msl_pipeline.constants.is_empty() {
                panic!("Supply pipeline constants via pipeline_constants instead of msl_pipeline.constants!");
            }
            let mut pipeline_options = params.msl_pipeline.clone();
            pipeline_options.constants = params.pipeline_constants.clone();
```